resources/party.ts

```diff
@@ -151,4 +167,36 @@ export default class PartyTracker {
   nameFromId(id: string): string | undefined {
     return this.idToName_[id];
   }
+
+  paramObjectFromName(
+    name: string,
+    defaultValue: keyof PartyMemberParamObject,
+    playerNicks: { [name: string]: string },
+  ): PartyMemberParamObject | undefined {
+    const partyMember = this.details.find((member) => member.name === name);
+    if (!partyMember)
+      return;
+
+    const jobName = Util.jobEnumToJob(partyMember.job);
+    const role = Util.jobToRole(jobName);
+    const ret: PartyMemberParamObject = {
+      id: partyMember.id,
+      job: jobName,
+      role: role,
+      name: name,
+      shortName: Util.shortName(name, playerNicks),
+    };
+
+    // Need to assign this afterwards so it can reference `ret`
+    ret.toString = () => {
+      const retVal = ret[defaultValue];
+      if (typeof retVal === 'string')
+        return retVal;
+      if (typeof retVal === 'number')
```

Comment:
Should we include boolean as well just for completeness?

Reply:
I left off boolean because boolean wasn't a valid type for the underlying params object.

Reply:
I guess my more general question was if it should be added there as well, or if you think that's an edge case that we won't use.